src/Test/Html/Internal/ElmHtml/ToString.elm

```diff
@@ -115,23 +115,25 @@ nodeRecordToString options { tag, children, facts } =
                     styleValues
                         |> List.map (\( key, value ) -> key ++ ":" ++ value ++ ";")
                         |> String.join ""
-                        |> (\styleString -> "style=\"" ++ styleString ++ "\"")
+                        |> (\styleString -> "style=\"" ++ escapeHtml styleString ++ "\"")
                         |> Just
 
         classes =
             Dict.get "className" facts.stringAttributes
-                |> Maybe.map (\name -> "class=\"" ++ name ++ "\"")
+                |> Maybe.map (\name -> "class=\"" ++ escapeHtml name ++ "\"")
 
         stringAttributes =
             Dict.filter (\k v -> k /= "className") facts.stringAttributes
                 |> Dict.toList
+                |> List.filter (\( k, _ ) -> not (isUnsafeName k))
                 |> List.map (Tuple.mapFirst propertyToAttributeName)
-                |> List.map (\( k, v ) -> k ++ "=\"" ++ v ++ "\"")
+                |> List.map (\( k, v ) -> k ++ "=\"" ++ escapeHtml v ++ "\"")
```

Comment:
This is perhaps the most important security fix of this PR.

Imagine you render user avatar images, and you would like to display the user name when hovering an avatar. You’d might do it like this:

```elm
Html.Attributes.attribute "title" userName
```

However, if the user chose the following name when they signed up:

```
"><script>alert("hacked")</script>
```

… then an alert box saying "hacked" will open on page load.

Now, it’s probably a good idea to validate user names a bit, but regardless of that it should be safe to shove any content into an attribute value, just like it’s safe to shove any content into `Html.text`.